src/components/Step1.tsx

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import { allCountries } from "./country";
+import {BsFillExclamationSquareFill}  from 'react-icons/bs'
+
+const validationSchema = yup.object().shape({
+  username: yup
+    .string()
+    .required("Username is required")
+    .min(4, "Username must be at least 4 characters")
+    .max(12, "Username must be at most 12 characters"),
+  email: yup
+    .string()
+    .required("Email is required")
+    .matches(
+      /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+      "Please enter a valid email address"
+    ),
+  phoneNumber: yup
+    .string()
+    .required("Phone Number is required")
+    .matches(
+      /^(?:\+?\d{1,3}\s?)?(?:\d{10}|\d{3}[-\.\s]?\d{3}[-\.\s]?\d{4})$/,
+      "Invalid phone number"
+    ),
+  country: yup
+    .string()
+    .required("Country is required")
+    .matches(
+      /^(?!.*\bSelect\b)/, // Use negative lookahead to ensure "SELECT" is not present
+      "Country is required"
+    ),
+});
+
+const mappedArray = allCountries.map(
+  (country) => `${country.code} ${country.name}`
+);
+// const mappedArray = allCountries.map(country => `<img src="${country.flag_1x1}" alt="${country.name} flag" /> ${country.code} ${country.name}`);
+
+type Props={
+  formData:any,
+  setFormData:any,
+  nextStep:any
+}
+
+const SignupPage = (props:Props) => {
+  const {formData,setFormData,nextStep}= props
+    const formik = useFormik({
+    initialValues: {
+      username: "",
+      email: "",
+      phoneNumber: "",
+      country: "",
+    },
+    initialErrors: {
+      username: "",
+    },
+    validationSchema,
+    onSubmit: (values) => {
+    const { username, email, phoneNumber, country } = values;
+    setFormData({ ...formData, username: username, email: email, phoneNumber: phoneNumber, country: country });
+    nextStep();
+      console.log("Form submitted with values:", values);
+    },
+  });
+
+  // const [Allow, setAllow] = useState(false);
+  // const textColorClass = Allow ? 'text-[#A3A0C0]' : 'text-[#000000]';
+  return (
+    <div className=" bg-gradient-to-b from-purple-50 via-blue-50 to-purple-50">
+      <div className="pb-10 absolute text-center items-center justify-center flex flex-col w-[100%] h-[30%]">
+        <h2 className="text-3xl font-semibold text-black mb-4 text-center">
+          Super test Form
+        </h2>
+        <h2 className="text-xl text-[#817DA4] mb-4 text-center">Initial Info</h2>
+      </div>
+      <div className="min-h-screen flex flex-row items-center justify-center">
+        <div className="relative -translate-x-[20%] -translate-y-[180%] h-[10%] w-[20%]">
+          <div className="flex flex-row items-center">
+            <div className="bg-[#594BD9] w-4 h-4 rounded"></div>
+            <div className="ml-2"> Initial Info</div>
+          </div>
```

Comment:
most of the code seems same let create a separate component for these? you thoughts?

Reply:
done